### DISCUSSION TITLE
Simu5G packet tracing

### DISCUSSION THREAD

**Discussion by Kavyashree14** [Unipisa/Simu5G#208] [Q&A]
Hello All,
I have tried to implement delay in the transmitted signal to simulate a scenario. With few KPIs like RLCDelay etc, I am unable to analyse the results. Could someone please guide me on how to trace the packets received by a particular user and then find out the effect of delay based on reception time? Any leads in this regard would be helpful.

Thanks in advance!

**Comment by alemilan** [comment]
Hello, 
I'm not sure if this could help you, but I've tried to do something similar to you. I inserted  a log (writing directly on a file) on every layer of the protocol stack, using the graphical components as a guide.
Using CBR packet as an example, maybe you could add some log at RLC layer, copying the packet and ''peekingAtFront' or 'atBack' so you can filter by destId or destAddress, compare the cbrHeader->getPayloadTimestamp() with simTime() for the specific IDframe. 

Hope this can be a path for you.

Best Regards.
